picocom.c

```diff
@@ -1616,6 +1616,10 @@ main(int argc, char *argv[])
 	init_history();
 #endif
 
+	r = term_get_baudrate(tty_fd, NULL);
+	if ( r != opts.baud )
+		fd_printf(STO, "WARNING: Desired baudrate is %d but applied baudrate is %d; this might depend upon your OS and/or hardware limitations.\r\n\r\n", opts.baud, r);
+
```

Comment:
I'm ok with issuing warnings when the software does not behave as expected. The problem is why only for the baudrate? Why not for databits, or stopbits, or parity, and so on (these settings could also be silently ignored by the h/w or driver). The driver could return an error when an unsupported setting is set (and some drivers, in some cases, do). In some cases the driver chooses to ignore the setting, silently. Picocom provides a method to see the actual setting and compare them with the requested ones. I think it's enough.